starknet/src/utils/types.cairo

```diff
@@ -115,33 +108,74 @@ impl ProposalStatusIntoU8 of Into<ProposalStatus, u8> {
     }
 }
 
-impl ProposalStatusPartialEq of PartialEq<ProposalStatus> {
-    fn eq(lhs: ProposalStatus, rhs: ProposalStatus) -> bool {
-        ProposalStatusIntoU8::into(lhs) == rhs.into()
+impl LegacyHashChoice of LegacyHash<Choice> {
+    fn hash(state: felt252, value: Choice) -> felt252 {
+        LegacyHash::hash(state, ChoiceIntoU8::into(value))
     }
+}
 
-    fn ne(lhs: ProposalStatus, rhs: ProposalStatus) -> bool {
-        ProposalStatusIntoU8::into(lhs) != rhs.into()
-    }
+#[derive(Option, Clone, Drop, Serde, StorageAccess)]
+struct Strategy {
+    address: ContractAddress,
+    params: Array<felt252>,
 }
 
-impl FinalizationStatusPartialEq of PartialEq<FinalizationStatus> {
-    fn eq(lhs: FinalizationStatus, rhs: FinalizationStatus) -> bool {
-        FinalizationStatusIntoU8::into(lhs) == rhs.into()
+impl PartialEqStrategy of PartialEq<Strategy> {
+    fn eq(lhs: @Strategy, rhs: @Strategy) -> bool {
+        lhs.address == rhs.address
+            && poseidon::poseidon_hash_span(
+                lhs.params.span()
+            ) == poseidon::poseidon_hash_span(rhs.params.span())
     }
 
-    fn ne(lhs: FinalizationStatus, rhs: FinalizationStatus) -> bool {
-        FinalizationStatusIntoU8::into(lhs) != rhs.into()
+    fn ne(lhs: @Strategy, rhs: @Strategy) -> bool {
+        !(lhs.clone() == rhs.clone())
     }
 }
 
+#[derive(Option, Clone, Drop, Serde)]
+struct IndexedStrategy {
+    index: u8,
+    params: Array<felt252>,
+}
+
+/// NOTE: Using u64 for timestamps instead of u32 which we use in sx-evm. can change if needed.
```

Comment:
What's the rationale for introducing this change?

Reply:
its the default type returned by the syscall, not much reasoning atm. We were not packing the timestamps anyway right now so felt unnecessary to cast